addons/sale/tests/test_sale_order.py

```diff
@@ -431,6 +431,13 @@ def test_tax_amount_rounding(self):
         })
         self.assertEqual(sale_order.amount_total, 15.41, "")
 
+    @users('public')
+    def test_public_user_order_locking(self):
+        self.sale_order.create_uid.groups_id += self.env.ref('sale.group_auto_done_setting')
+        self.sale_order.with_user(self.env.user.id).sudo().action_confirm()
+
+        self.assertEqual(self.sale_order.state, 'done')
```

Comment:
I am not sure I understand this correctly. Because after removing the decorator the self.env.user is not the public user anymore it is the odooBot and we are adding the group to it in ln 436. So, which user do you want to check that he does not have this group?

Reply:
That's because I wrote that comment before the next one... You can ensure `public_user` doesn't have the group though =]

Reply:
Done :slightly_smiling_face: